docs/docs/optimizing-performance.md

```diff
@@ -383,7 +383,7 @@ const y = x.set('foo', 'baz');
 x === y; // false
 ```
 
-In this case, since a new reference is returned when mutating `x`, we can safely assume that `x` has changed.
+In this case, since a new reference is returned when mutating `x`, we can safely assume that `x` has not changed.
```

Comment:
I see where the confusion is coming from, but I think this is also missing the point that's it's trying to communicate. When it says 

>  we can safely assume that `x` has changed.

It means we know the value that `x` was holding, which was updated in stored in a new reference `y`, has changed because the reference equality check failed. 

We should clarify this point better. I suggest something like:

> In this case, since a new reference is returned when mutating `x` we can use a reference equality check (`x === y`) to verify that the new value stored in `y` is different than the original value stored in `x`. 

It might also be useful to demonstrate that updating a record with the same value results in a passing equality check:

```
const y = x.set('foo', 'bar')
x === y // true
```